.github/workflows/packer-build-and-publish.yml

```diff
@@ -147,14 +152,68 @@ jobs:
           path: ${{ steps.build.outputs.readme }}
           overwrite: true
 
+  copy:
+    runs-on: ubuntu-latest
+    needs:
+      - check-if-main
+      - build
+    env:
+      SOURCE_ACCOUNT: "654654387067" # dev
+      DEST_ACCOUNT: "654654387067" # dev
+    steps:
+      - name: Authenticate to ${{ env.SOURCE_ACCOUNT }}
+        uses: grafana/shared-workflows/actions/aws-auth@main
+        with:
+          role-arn: "arn:aws:iam::${{ env.SOURCE_ACCOUNT }}:role/github-actions/packer-role"
+          pass-claims: "repository_owner, repository_name, job_workflow_ref, workflow_ref, event_name"
+          set-creds-in-environment: true
+
+      - name: Share AMIs with ${{ env.DEST_ACCOUNT }}
+        run: |
+          IFS=',' read -ra REGION_AMI_PAIRS <<< "${{ needs.build.outputs.amis }}"
+          for PAIR in "${REGION_AMI_PAIRS[@]}"; do
+            AMI_ID="${PAIR##*:}"
+            echo "Sharing AMI $AMI_ID with account ${{ env.DEST_ACCOUNT }}"
+            aws ec2 modify-image-attribute \
+              --image-id $AMI_ID \
+              --launch-permission "Add=[{UserId=${{ env.DEST_ACCOUNT }}}]"
+          done
+
+      - name: Authenticate to ${{ env.DEST_ACCOUNT }}
+        uses: grafana/shared-workflows/actions/aws-auth@main
+        with:
+          role-arn: "arn:aws:iam::${{ env.DEST_ACCOUNT }}:role/github-actions/packer-role"
+          pass-claims: "repository_owner, repository_name, job_workflow_ref, workflow_ref, event_name"
+          set-creds-in-environment: true
+
+      - name: Copy Shared AMI to Destination Account
+        run: |
+          IFS=',' read -ra REGION_AMI_PAIRS <<< "${{ needs.build.outputs.amis }}"
+          for PAIR in "${REGION_AMI_PAIRS[@]}"; do
+            AMI_ID="${PAIR##*:}"
+            AMI_REGION="${PAIR%%:*}"
+            echo "Copying $AMI_ID to $AMI_REGION on ${{ env.DEST_ACCOUNT }}"
+            NEW_AMI_ID=$(aws ec2 copy-image \
+              --region $AMI_REGION \
+              --source-image-id $AMI_ID \
+              --source-region $AMI_REGION \
+              --name "${{ needs.build.outputs.ami-name }}" \
+              --output text \
+              --query 'ImageId')
+          done
+
   publish:
     runs-on: ubuntu-latest
+    env:
+      BRANCH_NAME: update-readmes-${{ github.ref_name }}
     needs:
       - check-if-main
       - build
     steps:
       - name: Checkout repository
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0 # this is needed for the is-latest step
```

Comment:
There's no step with this ID so even though I'm suspicious about this I can't check what you mean. If you're talking about a step which runs `git` commands then it's usually possible to use `git fetch` or something similar to just download what we need, and save time on large repos.

Reply:
Done!